kotlin/workflow-core/src/main/java/com/squareup/workflow/WorkflowContext.kt

```diff
@@ -74,26 +77,25 @@ interface WorkflowContext<StateT : Any, in OutputT : Any> {
    * Ensures that the [channel][ReceiveChannel] returned from [channelProvider] is subscribed to, and
    * will send anything emitted on the channel to [handler] as a [ChannelUpdate].
    *
-   * This method ensures that only one subscription is active at a time for the given idempotence key,
-   * where idempotence keys are compared using [equals].
+   * This method ensures that only one subscription is active at a time for the given [type]+[key].
    * If this method is called in two or more consecutive [Workflow.compose] invocations with the same
    * key, [channelProvider] will only be invoked for the first one, and the returned channel will be
    * re-used for all subsequent invocations, until a [Workflow.compose] invocation does _not_ call this
    * method with an equal key. At that time, the channel will be [cancelled][ReceiveChannel.cancel],
    * with the assumption that cancelling the channel will release any resources allocated by the
    * [channelProvider].
    *
-   * **This method is intended for library authors**, since a unique idempotence key must be specified
-   * explicitly. Most code will want to call one of the variants that accepts a [String] `key`
-   * parameter (i.e. [onReceive]). See [IdempotenceKey] for factory methods to derive keys from types.
-   *
+   * @param type The [KType] that represents both the type of data source (e.g. `ReceiveChannel` or
+   * `Deferred`) and the element type [E].
```

Comment:
```
e.g. `KTypes.fromGenericType(ReceiveChannel::class, E::class)
```

Reply:
Is there any way that we can enforce that `E` matches this? Assuming not, just thought I'd ask.

Reply:
Nope :(